File: cpp/source/event/output/ChannelsOpened.cpp
#include "event/output/ChannelsOpened.hpp"

#include <sstream>
#include <iterator>

using namespace fm;
using namespace fm::event::output;

ChannelsOpened::ChannelsOpened(std::shared_ptr<std::vector<std::shared_ptr<traffic::IChannel>>> _channels) :
    FacadeEvent(CHANNELS_OPENED),
    channels(_channels)
{
}

const std::vector<std::shared_ptr<traffic::IChannel>>& ChannelsOpened::getChannels() const
{
    return *channels;
}

std::string ChannelsOpened::toString() const
{
    std::ostringstream oss;
    oss << "[";
    for (std::shared_ptr<traffic::IChannel> c : *channels)
    {
        oss << c->getId() << ",";
    }
    oss << "]";
    return "CHANNELS_OPENED: " + oss.str();
}

File: java/src/main/java/com/fleetmgr/sdk/client/event/output/facade/ChannelsOpened.java
package com.fleetmgr.sdk.client.event.output.facade;

import com.fleetmgr.sdk.client.traffic.Channel;

import java.util.List;

/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class ChannelsOpened extends FacadeEvent {

    private List<Channel> channels;

    public ChannelsOpened(List<Channel> channels) {
        super(Type.CHANNELS_OPENED);
        this.channels = channels;
    }

    public List<Channel> getChannels() {
        return channels;
    }

    @Override
    public String toString() {
        return "ChannelsOpened{" +
                "channels=" + channels +
                '}';
    }
}
